client/modules/IDE/components/SketchList.jsx

```diff
@@ -9,12 +9,19 @@ import { bindActionCreators } from 'redux';
 import * as ProjectActions from '../actions/project';
 import * as SketchActions from '../actions/projects';
 import * as ToastActions from '../actions/toast';
+import orderListByDate from '../../../selectors/Sort-Sketches';
 
 const trashCan = require('../../../images/trash-can.svg');
+const arrowUp = require('../../../images/arrow5_up.svg');
+const arrowDown = require('../../../images/arrow5_down.svg');
 
 class SketchList extends React.Component {
   constructor(props) {
     super(props);
+    this.state = {
```

Comment:
why is this in the state and not in the redux store?

Reply:
This is in the redux store but I had to show svgs based on createdAt and updateAt fields and didn't wanted to store both in the store, now the store just has order and orderbyy

Reply:
i'm still confused as to why this is stored in the component state—isn't just the order and order by enough?

Reply:
I'll try to explain this.
I have put in just 2 things in the redux store , order and orderBy , Say for example I click on `createdAt` and `Ascending` and then the redux store values change to order = Ascending and orderBy = Descending , Now I click on `updatedAt` and `Ascending`  and now the redux store values change to orderBy = updatedAt and order = `Ascending` , now I have to render svgs based on the current order of createdAt , but redux store holds values of updatedAtorder , so there is no way of knowing what is the current order of updatedAt.
There can be 2 solutions for this.
1. Making isolated state.
2. Maintaing local state (which I have done in this case)

Reply:
i'm not sure i totally understand what you mean here. you should only ONE field, ONE direction at a time. The expected behavior is:
1. By default, the sort order is "Created Date, Descending". ONLY the "Created Date" field, since it is selected, renders the "ASC/DSC" arrows
2. If you click on another field, then simply apply a default value to order (maybe ASC? maybe DSC if it's date, ASC if it's a string?)